src/test/crimson/seastore/transaction_manager_test_state.h

```diff
@@ -244,26 +246,27 @@ class TMTestState : public EphemeralTestState {
 
   TMTestState(std::size_t num_devices) : EphemeralTestState(num_devices) {}
 
-  virtual void _init() override {
+  virtual seastar::future<> _init() override {
     auto sec_devices = devices->get_secondary_devices();
     auto p_dev = devices->get_primary_device();
     tm = make_transaction_manager(p_dev, sec_devices, true);
     epm = tm->get_epm();
     lba_manager = tm->get_lba_manager();
     cache = tm->get_cache();
+    return seastar::now();
   }
 
-  virtual void _destroy() override {
+  virtual seastar::future<> _destroy() override {
     epm = nullptr;
     lba_manager = nullptr;
     cache = nullptr;
     tm.reset();
+    return seastar::now();
   }
 
   virtual seastar::future<> _teardown() {
     return tm->close().safe_then([this] {
-      _destroy();
-      return seastar::now();
+      return _destroy();
```

Comment:
Seems not needed.

Reply:
need call shard_stores.stop() in _destroy(), so it should return seastar::future<>.

Reply:
I meant in the parent implementation, `EphemeralTestState::_destroy()` is already called after `EphemeralTestState::_teardown()`, right?

Reply:
right, fix it.